### DISCUSSION TITLE
Incorrect footnote generation position

### DISCUSSION THREAD

**Discussion by jdysya** [vuepress-theme-hope/vuepress-theme-hope#3110] [Q&A] [answered]
I have tried to realize the function related about the issue([能否支持obsidian的一些链接语法 #3069](https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/3069)).
Now, I have reached something:
### 1. To  convert the ```ad-xxx``` to :::xxx ::: automatically
<img width="1070" alt="image" src="https://user-images.githubusercontent.com/73117213/236453916-393657ab-8fe0-4400-a989-26572ae966d8.png">
<img width="860" alt="image" src="https://user-images.githubusercontent.com/73117213/236453997-aaf2792e-9cda-4898-9229-8e729ee6bbf5.png">

### 2. To convert Tag syntax in obsidian to badge in vuepress automatically.
<img width="1054" alt="image" src="https://user-images.githubusercontent.com/73117213/236454213-8d6aafab-7db0-4658-89db-7b7b1baab952.png">
<img width="862" alt="image" src="https://user-images.githubusercontent.com/73117213/236454313-8879b25d-c7c7-488b-8c49-6ae3332734f8.png">

### The specific implementation method is as follows(modify the `config.js`):

```js
extendsMarkdown: (md) => {
    const defaultFenceRenderer = md.renderer.rules.fence; // 保存默认的 fence 渲染规则
    // 匹配 #tag 格式的标签
    const tagRegex = /#[\u4e00-\u9fa5\w]+/g;
    md.renderer.rules.fence = function (tokens, idx, options, env, self) {
      const token = tokens[idx];
      if (token.tag === "code") {
        const info = token.info.trim();
        if (info.startsWith("ad-")) {
          // 处理特定的代码块
          const type = info.slice(3);
          return `<div class="hint-container ${type}"><p class="hint-container-title">${type}</p>
            ${md.render(token.content, env)}
            </div>`;
        }
      }
      // 按照默认规则处理代码块
      return defaultFenceRenderer(tokens, idx, options, env, self);
    };
    md.renderer.rules.text = (tokens, idx) => {
      let text = tokens[idx].content;

      // 对HTML特殊字符进行转义
      text = md.utils.escapeHtml(text);

      let matches = text.match(tagRegex);

      if (matches) {
        matches.forEach((match) => {
          // 提取标签名称
          let tagName = match.slice(1);

          // 生成 <Badge> 标签
          let badge = `<Badge type="danger">${tagName}</Badge>`;

          // 替换原始文本中的标签
          text = text.replace(match, badge);
        });
      }

      return text;
    };
  },
```

However, the tests found some problems with where the footnotes were generated：
<img width="901" alt="image" src="https://user-images.githubusercontent.com/73117213/236455201-cca8ed4c-dda3-4400-9402-e2bd1a10c35d.png">
Almost every container generates a footnote element. **I think it's because I parsed the documents in the container too soon**
Since I'm not familiar with the implementation details, anyone please tell me how to modify the code.

**Comment by Mister-Hope** [comment, marked as answer]
You shall not call `md.render` unless you want a new document there. Try to parse them as standard container tokens.

You may need to modify the fence rule to let it understand that fence using some special info should be parse to other tokens, in your case is container.

That's why I am not interested in implementing it. It's totally weird to wrap markdown content with fence as they can even not being correctly highlighted.
